src/hotspot/cpu/aarch64/aarch64.ad

```diff
@@ -16551,6 +16605,30 @@ instruct array_equalsC(iRegP_R1 ary1, iRegP_R2 ary2, iRegI_R0 result,
   ins_pipe(pipe_class_memory);
 %}
 
+instruct arrays_hashcode(iRegP_R1 ary, iRegI_R2 cnt, iRegI_R0 result, immI basic_type,
+                         vRegD_V0 vtmp0, vRegD_V1 vtmp1, vRegD_V2 vtmp2, vRegD_V3 vtmp3,
+                         vRegD_V4 vtmp4, vRegD_V5 vtmp5, vRegD_V6 vtmp6, vRegD_V7 vtmp7,
+                         vRegD_V12 vtmp8, vRegD_V13 vtmp9, vRegD_V14 vtmp10,
+                         vRegD_V15 vtmp11, vRegD_V16 vtmp12, vRegD_V17 vtmp13,
+                         rFlagsReg cr)
```

Comment:
Using fixed registers here is rather odd.

Is there some reason not simply to use `vReg` here, rather than named specific vector registers? You could just pass them down to `arrays_hashcode`. This issue isn't drop-dead-critical, but it would simplify this patch, which is otherwise fine.

Reply:
> Using fixed registers here is rather odd.

My mistake. I see that you're calling a stub, unlike x86 which expands inline. It could go either way, whichever you choose to do is OK. Inline might be a bit more performant, but I suspect it's marginal.

Reply:
The implementation is split into two parts: an unrolled scalar loop that handles up to tail 16/32 elements depending on the data type and a vectorized Neon loop. Inlining both parts takes more than 300B of code cache. Most of that size accounts for the Neon loop so it was moved into a stub. The scalar loop expands inline, the Neon loop is implemented via a stub which is called from the inlined part. This had no statistically significant effect on the performance of  `ArraysHashCode` and `StringHashCode` benchmarks compared to the fully inlined (no stub) version.